Sources/SwiftGenKit/Parsers/Strings/StringsDictEntry.swift

```diff
@@ -0,0 +1,212 @@
+
+      case zero, one, two, few, many, other
+    }
+  }
+}
+
+// - MARK: Decodable conformance
+
+extension StringsDict: Decodable {
+  // Custom CodingKeys struct to be able to parse dictionaries with dynamic keys.
+  // This is necessary, because each variable used in the formatKey is used as the key
+  // to another dictionary containing the rules for that variable.
+  private struct CodingKeys: CodingKey {
+    var intValue: Int?
+    var stringValue: String
+
+    init?(intValue: Int) {
+      self.intValue = intValue
+      self.stringValue = "\(intValue)"
+    }
+    init?(stringValue: String) {
+      self.stringValue = stringValue
+    }
+    init(key: String) {
+      self.stringValue = key
+    }
+
+    // fixed keys
+    static let formatKey = CodingKeys(key: "NSStringLocalizedFormatKey")
+    static let variableWidthRules = CodingKeys(key: "NSStringVariableWidthRuleType")
+  }
+
+  init(from coder: Decoder) throws {
+    let container = try coder.container(keyedBy: CodingKeys.self)
+    let formatKey = try container.decodeIfPresent(String.self, forKey: .formatKey)
+    let variableWidthRules = try container.decodeIfPresent([String: String].self, forKey: .variableWidthRules)
+
+    // We either have a formatKey OR we have a variableWidthRule.
+    switch (formatKey, variableWidthRules) {
+    case (.none, .none), (.some, .some):
+      throw Strings.ParserError.invalidFormat
+    case let (.some(formatKey), .none):
+      let variables = try StringsDict.decodeVariableRules(in: container, formatKey: formatKey)
+      self = .pluralEntry(PluralEntry(formatKey: formatKey, variables: variables))
+    case let (.none, .some(variableWidthRules)):
+      self = .variableWidthEntry(VariableWidthEntry(rules: variableWidthRules))
+    }
+  }
+
+  private static func decodeVariableRules(
+    in container: KeyedDecodingContainer<StringsDict.CodingKeys>,
+    formatKey: String
+  ) throws -> [PluralEntry.Variable] {
+    guard let variableKeyTuples = StringsDict.variableKeysFromFormatKey(formatKey) else { return [] }
+    let sortedVariableKeys = variableKeyTuples
+      .sorted { lhs, rhs in
+        // Sort by positional argument if present, otherwise by occurrence in the format key
+        switch (lhs.positionalArgument, rhs.positionalArgument, lhs.range.lowerBound, rhs.range.lowerBound) {
+        case let (.none, .none, lhs, rhs):
+          return lhs < rhs
+        case (.some, .none, _, _):
+          return true
+        case (.none, .some, _, _):
+          return false
+        case let (.some(lhs), .some(rhs), _, _):
+          return lhs < rhs
+        }
+      }
+      .map { $0.key }
+
+    return try sortedVariableKeys.reduce(into: [PluralEntry.Variable]()) { variables, variableKey in
+      let variableRule = try container.decode(PluralEntry.VariableRule.self, forKey: CodingKeys(key: variableKey))
+      variables.append(PluralEntry.Variable(key: variableKey, rule: variableRule))
+    }
+  }
```

Comment:
nit: I think this whole method should be moved a couple of lines below, in the `// MARK: Helper` section (and could also benefit a code comment)

Reply:
I'd like to keep it here, because this is still part of the decoding.

Reply:
Mmmmg but the `variableKeysFromFormatKey`, which is called by this `decodeVariableRules`, is also part of the decoding but is already in the "Helper" section… 🤷

Reply:
True, then I'll rename the method and extract the `try container.decode(...)` from it into the initializer above.